### PR TITLE
Fix 195

### PR DESCRIPTION
**Pull request by oneminch**
An attempt to fix incognito mode cache overwriting

### CODE REVIEW

File: packages/extension/src/common/cache.js [hunk 1]
```diff
@@ -7,7 +9,11 @@ export const FIRST_INSTALL_KEY = 'firstInstall';
 export const LAST_COMMENT_KEY = 'lastComment';
 
 export function setCache(key, value) {
-  return browser.storage.local.set({ [key]: value });
+  isPrivateMode().then(function(isPrivate) {
```
Comment (idoshamun) [inline packages/extension/src/common/cache.js:12]: You need to return the promise here.
Otherwise, it will not be usable from the outside

**Review by idoshamun — CHANGES_REQUESTED** [review]
Hey @oneminch,

Thanks for submitting another PR 🙌
First of all, I think that you have to set up a local environment so you can test your PR as it gets more complex slowly.
Second, make sure to run the linter locally so you can fix the lining issues.
The last and most important point is that the `setCache` method is called many times and the `isPrivateMode` function can be a little heavy. My suggestion is to cache the result of this function as it should remain the same across the session of the user.

Let me know what do you think and if you need any further guidance.

**Comment by oneminch** [comment]
I spent over an hour on setting up the environment locally but I had issues with Docker on my device and I get errors when I run the `docker-compose` setup commands. That's why I submitted a PR without testing. I'll have another go at fixing the other issues you mentioned.

**Comment by idoshamun** [comment]
It is much appreciated 🙏

**Comment by oneminch** [comment]
Do you know what could be causing this? It gets stuck at this point and I couldn't go further setting up the local environment. I already did `docker-compose pull`. The problem is when I run `docker-compose up`.
docker

File: packages/extension/src/common/cache.js (hunk 1 above)
Comment (oneminch) [reply packages/extension/src/common/cache.js:12]: What do you mean by that? Do you mean outside the if statement?

**Comment by idoshamun** [comment]
I'm glad to see that you've sorted out the docker-compose installation issue! So great :)
I'll check out this new issue later on and let you know.

**Comment by idoshamun** [comment]
Hey @oneminch,

I created a clean machine and tried to run `docker-compose up` and it works perfectly fine.
Please make sure you run `docker-compose up` from the root folder of `daily-apps`.
If it still doesn't work, can you share some information about your environment? Such as Operation System, Docker version, docker-compose version.

**Comment by oneminch** [comment]
OS: Windows 10 Home
Docker version: 19.03.13
docker-compose version: 1.27.4

**Comment by idoshamun** [comment]
Honestly, I've no experience with Docker (or docker-compose) on Windows.
But according to the logs, it seems that docker-compose didn't mount `wait-for.sh` into the container although it should.
The file is located in the root directory of `daily-apps`.
You can see the setting here:
https://github.com/dailydotdev/daily-apps/blob/master/docker-compose.yml#L27

After installing Docker and docker-compose you shouldn't do any further steps to configure it. At least on Mac and Linux.

**Comment by oneminch** [comment]
I have been running it from the root directory. Probably it's the relative address of `wait-for.sh` that created this issues.
You should go ahead and assign this issue to someone else. This isn't as beginner-friendly as I thought. Thanks for the help!

**Comment by idoshamun** [comment]
Sorry about that.
Unfortunately, docker is still a few steps behind on Windows machines.
I have to check how to make it easier to set up a local environment on Windows. 
Thanks for everything!

**Pull request closed without merging**